lib/ansible/plugins/lookup/first_found.py

```diff
@@ -106,71 +110,100 @@
 
 from jinja2.exceptions import UndefinedError
 
-from ansible.errors import AnsibleFileNotFound, AnsibleLookupError, AnsibleUndefinedVariable
+from ansible.errors import AnsibleLookupError, AnsibleUndefinedVariable
+from ansible.module_utils.common._collections_compat import Mapping, Sequence
 from ansible.module_utils.six import string_types
-from ansible.module_utils.parsing.convert_bool import boolean
 from ansible.plugins.lookup import LookupBase
 
 
+def _split_on(terms, spliters=','):
+
+    # TODO: fix as it does not allow spaces in names
+    termlist = []
+    if isinstance(terms, string_types):
+        for spliter in spliters:
```

Comment:
Can this happen at all, since you added `type: list`?

Reply:
not for the config route, but yes for the 'inline dict' .. again keeping backwards compatibility here.

Reply:
But the inline dict route now also uses the config manager.

Reply:
sorry, inline string, not dict .. this lookup allowed for many bad ways of specifying params

Reply:
Hmm, I still don't see where that should happen :) Right now, `_split_on` is only called on the results of `self.get_option('files')` and `self.get_option('paths')`, so only for data that went through ConfigManager.

Reply:
they can still be inline strings at that point, since it was allowed before, it is only way to continue to allow for that

files and path CAN be a list or can be char separated string, in the case of paths 2 chars, in files 3

Reply:
```- hosts: localhost
  gather_facts: false
  tasks:
  - name: test legacy formats
    set_fact: hatethisformat={{item}}
    vars:
      params:
        files: not/a/file.yaml;hosts
        paths: not/a/path:/etc
      skip: True
    loop: "{{ q('first_found', params) }}"```